---
title: "Surface_Water_OKG_comparison"
output: html_document
date: "2023-10-02"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries}
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(vegan))
suppressPackageStartupMessages(library(dplyr))
suppressPackageStartupMessages(library(SRS))
suppressPackageStartupMessages(library(patchwork))
suppressPackageStartupMessages(library(ggfortify))
suppressPackageStartupMessages(library(ggiraphExtra))
suppressPackageStartupMessages(library(ggpubr))
suppressPackageStartupMessages(library(ALDEx2))
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(readr))
suppressPackageStartupMessages(library(DESeq2))
suppressPackageStartupMessages(library(naniar))
suppressPackageStartupMessages(library(factoextra))
```

```{r define colors throughout the script}
# When one uses the same categores throughout the paper, it can be easy to set the colours here
# When plotting, run for example scale_colour_manual(values = cols.phylum)

#Bays

cols.bay <- c(
  "heated" = "orange",
  "control" = "blue"
)

#16S amplicon data

cols.phylum <- c(
  "Actinobacteriota"="#7ab5bc",
  "Bacteroidota" = "#a6cee3",
  "Cyanobacteria" = "#b2df8a",
  "Proteobacteria" = "#fb9a99",
  "Others"="grey"
  )

cols.family<- c(
  "AcAMD-5" ="#84afde",
  "Burkholderiaceae"="#7ab5bc",
  "Crocinitomicaceae" ="#a2abc9",
  "Cyclobacteriaceae" ="#e3fffa",
  "D2472" = "#a6cee3",
  "Flavobacteriaceae"=  "#1f78b4",
  "Ilumatobacteraceae"="#73cfc6",
  "Methylophilaceae" = "#ffffcd",
  "Microbacteriaceae" = "lightgoldenrod", 
  "Mycobacteriaceae" = "#b2df8a",
  "Nanopelagicaceae" = "yellowgreen",
  "Nostocaceae" = "#33a02c",
  "Porticoccaceae" = "olivedrab",
  "Rhodobacteraceae" = "#9eb383",
  "S36-B12" = "lemonchiffon3",
  "Schleiferiaceae" = "lightpink", 
  "SHWZ01" ="#ddcaff",
  "Sphingomonadaceae" ="#d49abe",
  "Spirosomaceae" = "#db9d88",
  "UBA5976" ="#bea4b6",
  "UBA8139"=   "#fb9a99",
  "UBA9320"= "#f8cea0"
  )


##mRNA

cols.mRNA.phylum <- c(
    "Cyanobacteria" = "#84afde",
  "Delongbacteria" = "#7ab5bc",
  "Methylomirabilota"= "#a2abc9",
  "Micrarchaeota"= "#1f78b4",
  "Zixibacteria"= "#ffffcd",
  "Desulfobacterota_B"= "#b2df8a",
  "Hydrogenedentota"= "#33a02c",
  "DTU030" ="#9eb383",
  "Elusimicrobiota" ="#ddcaff",
  "Firmicutes_F" = "#d49abe",
  "UBA10199" = "#db9d88",
  "WOR-3" = "#fb9a99",
  "Desulfobacterota_D" ="#f8cea0",
  "Others" ="grey"

)

#Family

cols.mRNA.family<- c(
  "Actinomycetia uncl. uncl" ="#84afde",
  "CADAEX01"="#7ab5bc",
  "Chromatiaceae" ="#a2abc9",
  "Chroococcidiopsidaceae" ="#e3fffa",
  "Desertifilaceae" = "#a6cee3",
  "Elainellaceae"=  "#1f78b4",
  "JA-3-3Ab"="#73cfc6",
  "JAAUUE01" = "#ffffcd",
  "JAAYBP01" = "lightgoldenrod", 
  "JACDCP01" = "#b2df8a",
  "Ketobacteraceae" = "yellowgreen",
  "Leeiaceae" = "#33a02c",
  "Methylomirabilaceae" = "olivedrab",
  "Nevskiales uncl." = "#9eb383",
  "Oscillatoriaceae_A" = "lemonchiffon3",
  "Rickettsiales uncl." = "lightpink", 
  "SHVT01" ="#ddcaff",
  "Sphaerochaetaceae" ="#d49abe",
  "Sulfurifustaceae" = "#db9d88",
  "UBA10165" ="#bea4b6",
  "UBA7541"=   "#fb9a99",
  "Others" ="grey"
  )

# Need inspiration for you colour schemes? https://www.datanovia.com/en/blog/top-r-color-palettes-to-know-for-great-data-visualization/
# Or, https://colorbrewer2.org/#type=sequential&scheme=BuGn&n=3
```

```{r Tables - 16S amplicon sequencing}
#meta 
meta <- read.table("16S_ampliseq/meta.txt", stringsAsFactors = F,fileEncoding="latin1", sep="\t", header=T)
meta$group <- factor(paste0(meta$bay, meta$filter))

#counts 
counts <- read.table("16S_ampliseq/ASV_table.tsv", stringsAsFactors = F, header =T)%>%
          gather(sample,count, 2:ncol(.))%>%
          filter(count > 0)

#taxa
taxa <- read.delim("16S_ampliseq/ASV_tax_species.tsv",stringsAsFactors = FALSE, header=TRUE)%>%
replace_with_na_if(.predicate = is.character,
                     condition = ~.x %in% (""))%>%
        replace_na(list(Domain="zz_Unclassified"))%>%
        replace_na(list(Kingdom="zz_Unclassified"))%>%
        replace_na(list(Phylum="zz_Unclassified"))%>%
        replace_na(list(Class="zz_Unclassified"))%>%
        replace_na(list(Order="zz_Unclassified"))%>%
        replace_na(list(Family="zz_Unclassified"))%>%
        replace_na(list(Genus="zz_Unclassified"))%>%
        replace_na(list(Species="zz_Unclassified"))%>%
        replace_na(list(Species_exact="zz_Unclassified"))

#Phytoref blast
phytoref.16S <- read.delim("16S_ampliseq/ASV_seqs_blast_nt.tsv", stringsAsFactors = F, header=F)%>%
  dplyr::rename(ASV_ID = V1)%>%
  group_by(ASV_ID)%>%
  filter(row_number()==1)%>%
  ungroup()

#Combine
Tab_all <- counts%>% 
  left_join(meta, by="sample") %>% 
  left_join(taxa, by="ASV_ID")%>% 
  filter(Family != "Mitochondria") %>% 
  filter(Order  != "Chloroplast") 

#Looking for the unknown cyanobacteria
tab.cyano <- Tab_all %>%
  filter(Phylum %in% "Cyanobacteria")%>%
  filter(Family %in% "zz_Unclassified")%>%
  left_join(phytoref.16S, by="ASV_ID")

#filter  out eukaryotic asvs
Tab_all <- Tab_all%>%
  anti_join(tab.cyano, by="ASV_ID")%>%
  left_join(phytoref.16S, by="ASV_ID")
```

```{r Format tables - 16S amplicon sequencing}
#raw counts wide format
countsRAW <- Tab_all %>%  
  group_by(sample) %>% 
  # Filter to keep only rows from samples having at least 1000 counts
  filter(sum(count) >= 1000) %>% 
  dplyr::select(ASV_ID, sample, count) %>% #select ISV, sample and count
  spread(ASV_ID,count, fill= 0) %>% #wide format with sequence and count, filling gaps with 0
  remove_rownames() %>% #remove of row names
  column_to_rownames(var = "sample")

#relative abundance calculation
Tab_all <- Tab_all %>%
  # Group by sample, to get access to the group wise sum of counts
  group_by(sample) %>% 
  # Filter to keep only rows from samples having at least 1000 counts
  filter(sum(count) >= 1000) %>% 
  # Since we have the data grouped, we can calculate the relative abundance here
  mutate(relab = count/sum(count)) %>%
  ungroup()

#wide format
widecount<- Tab_all %>% 
  dplyr::select(ASV_ID, sample, count) %>% 
  spread(ASV_ID,count, fill= 0) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "sample")

#wide format relab
widerelab<- Tab_all %>% 
  dplyr::select(ASV_ID, sample, relab) %>% 
  spread(ASV_ID,relab, fill= 0) %>%
  remove_rownames() %>% 
  column_to_rownames(var = "sample")


```

```{r rarefaction curve - 16S amplicon sequencing}

#color according to the bays
col <- c ("blue", "orange2")
grp <- factor(meta$bay, levels= c("control","heated"))
cols <- col[grp]

(raremax <- min(rowSums(countsRAW)))

#Rarefaction curve plot
out <-  rarecurve(countsRAW, step = 20, sample = raremax, col = cols , cex = 0.6)

```

```{r Alpha Diversity - 16s amplicon sequencing}
#SRS analysis
example_input_data <- Tab_all %>%
  dplyr::select(ASV_ID, sample, count)%>%
  spread(sample,count, fill= 0) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "ASV_ID")

#(e.g. species counts of the library with the lowest sequencing depth):
Cmin <- min(colSums(example_input_data))
Cmin

SRS_output <- SRS(data = example_input_data, Cmin = Cmin)
SRS_output

#Shannon Diversity Index
SRSshannon<- SRS_output %>%
  t()%>%
  data.frame()%>% 
  rownames_to_column(var="sample")%>% 
  plyr::ddply(~sample, function(x) {vegan::diversity(x[-1], index="shannon")}) %>%
  dplyr::rename(shannon= V1) %>% #Shannon 
  left_join(meta, by="sample")

SRSshannon %>%
  group_by(bay)%>%
  summarise(mean=sd(shannon))%>%
  ungroup()

##Plot
shanSRS_<- SRSshannon %>%
   group_by(bay)%>%
  mutate(mean=mean(shannon))%>%
  mutate(sd=sd(shannon))%>%
  ungroup()%>%
 ggplot(aes(x = bay, y = mean, color=bay)) +
  geom_point(size=3) +
 #geom_line(aes(group=filter))+
   geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.3,
                 position=position_dodge(.05)) +
 # geom_smooth(method="lm",aes( group=bay, color=bay))+
scale_color_manual(values=cols.bay)+
  guides(size=guide_legend(order=2),
         shape= guide_legend(override.aes=list(size=2),
        colour=guide_legend(override.aes=list(size=4))))+
 labs(x = "bay", y = "Shannon Index") +

   theme_classic()+  theme(
    legend.position = 'bottom'
  )



#non-parametric test to check for significant differences between bays
krusk <-kruskal.test(shannon ~ bay, data = SRSshannon)
```

```{r Overview Phylum & Family - 16S amplicon sequencing}
#Phylum
phyl.16 <- Tab_all %>%
  group_by(sample)%>%
filter(relab >= 0.005) %>%
  ungroup()%>%
ggplot(aes(x = name2, y = relab, fill = Phylum)) +
geom_bar( stat="identity") +
      scale_fill_manual(values=cols.phylum)+
coord_flip() +
theme(
    legend.position = 'bottom'
  )+
  xlab('bay (n=6)') + ylab('relative abundance >0.5%') +
   theme_classic()+
  theme(
    legend.position = 'bottom',
    axis.text.x = element_text(angle = 60, hjust = 1)
  )+
  ylim(0,1)


#Overview abundances of phyla
Tab_all %>%
  group_by(sample)%>%
filter(relab >= 0.005) %>%
  ungroup()%>%
  group_by(bay, Phylum,sample)%>%
  summarise(relab=sum(relab))%>%
    summarise(relab=mean(relab))%>%
  ungroup()

#Family
fam.16s <-Tab_all %>%
  group_by(sample)%>%
filter(relab >= 0.01) %>%
  ungroup()%>%
ggplot(aes(x = name2, y = relab, fill = Family)) +
geom_bar(stat="identity") +
coord_flip() +
        scale_fill_manual(values=cols.family)+
theme(
    legend.position = 'bottom'
  )+
  xlab('bay (n=6)') + ylab('relative abundance >1%') +
   theme_classic()+
  theme(
    legend.position = 'right',
    axis.text.x = element_text(angle = 60, hjust = 1)
  )+
  ylim(0,1)
```

```{r Differential abundance analysis ALDEx2 - 16S amplicon sequencing}
set.seed(254)#Differential abundance analysis cannot be done on higher level, always on ASV level

Aldex.bay <- Tab_all %>%
  group_by(sample, ASV_ID)%>%
  summarise(count = sum(count))%>%
  ungroup()%>%
 dplyr::select(ASV_ID, sample, count)%>%
  spread(sample, count, fill=0)%>%
  column_to_rownames("ASV_ID")%>%
  ALDEx2::aldex(conditions = meta$bay)%>%
  tibble::rownames_to_column("ASV_ID")%>%
  inner_join(taxa, by="ASV_ID")

# Bland-Altman or MA plot (left).It shows the association between the relative abundance and the magnitude of the difference per sample. 
par(mfrow = c(1, 2))
  aldex.plot(
    Aldex.bay, 
    type = "MA", 
    test = "welch", 
    xlab = "Log-ratio abundance",
    ylab = "Difference",
    cutoff = 0.05
  )
#shows the dispersion on the x-axis instead of log-ratio abundance. Red dots represent genera that are differentially abundant (q≤0.1 ) between the 2 groups. Black points are rare taxa and grey ones are abundant taxa. The dashed line represent an effect size of 1. See Gloor, Macklaim, and Fernandes (2016) to learn more about these plots.  
    aldex.plot(
    Aldex.bay, 
    type = "MW", 
    test = "welch",
    xlab = "Dispersion",
    ylab = "Difference",
    cutoff = 0.05
  )
   

#Filter for significant DA and effect sizes over 1/-1

  Aldex.sig.ASV.map <- Aldex.bay%>%
  filter(wi.eBH <= 0.05)  %>% # here we chose the wilcoxon output rather than tt
  dplyr::select(ASV_ID, we.eBH, wi.eBH, effect, overlap) %>% distinct()%>%filter(!between(effect, -1, 1))%>%left_join(taxa, by="ASV_ID")
#Include a new column with either heated or control
  Aldex.sig.ASV.col.map <- Aldex.sig.ASV.map %>% 
  mutate(bay = ifelse(effect>0, "heated", "control"))
  
  
  #Effect size and overlap
  
  par(mfrow=c(1,2))
plot(Aldex.bay$effect, Aldex.bay$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Effect size", ylab="P value", main="Effect size plot")
points(Aldex.bay$effect, Aldex.bay$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")
legend(15,1, legend=c("P value", "BH-adjusted"), pch=19, col=c("blue", "red")) 

plot2 <- plot(Aldex.bay$diff.btw,Aldex.bay$we.ep, log="y", cex=0.7, col=rgb(0,0,1,0.2),
  pch=19, xlab="Difference", ylab="P value", main="Volcano plot")
points(Aldex.bay$diff.btw, Aldex.bay$we.eBH, cex=0.7, col=rgb(1,0,0,0.2),
  pch=19)
abline(h=0.05, lty=2, col="grey")

#blue p-value / red BH-adjusted p-value


#Heatmap of the Da of interest summarized on Family level  
 DA.fam.16S <- Aldex.sig.ASV.col.map %>%
    filter(!Family %in% "zz_Unclassified")%>%
group_by(Family, bay)%>%
    mutate(effect =mean(effect))%>%
    ungroup()%>%   # First sort by val. This sort the dataframe but NOT the factor levels
    ggplot( aes(factor(Family, 
       levels = rev(levels(factor(Family)))),bay, fill=effect))+
    geom_tile()+
    coord_flip()+
    scale_x_discrete(limits = rev(levels(Aldex.sig.ASV.col.map$Family)))+
    scale_fill_gradient2(low = "blue",
                  mid="white",
                      high = "orange",
                      guide = "colorbar",
                  limits=c(-10,10))+
      theme_classic()+
     theme(axis.text.x = element_text(angle = 45, hjust=1),
           panel.background = element_rect(fill="white", color="black"),
                                           legend.position = 'bottom')+
     xlab('Family') + ylab('bay (n=6)')
   
```

```{r PCA - 16S amplicon sequencing}
##PCoA with cmdscale 

## Distance matrix based on Bray Curtis dissimilarities
BC.dist=vegdist(widerelab, distance="bray")

pca <- prcomp(t(BC.dist))
pca$sdev
get_eigenvalue(pca)
fviz_eig(pca)

#PerMANOVA
perma <- adonis2(BC.dist ~bay, data=meta, permutations=999)

#PCA
df <- cbind(meta, pca$x)
pca.16 <- ggplot(df,aes(x=PC1, y=PC2, color = bay)) + geom_point(size=5)+
   scale_color_manual(values=cols.bay)+
   theme_classic()+
  theme(
    legend.position = 'bottom'
  )

#Proportion explained 
eigs <-df$sdev^2
eigs[1] / sum(eigs)

rbind(
  SD = sqrt(eigs),
  Proportion = eigs/sum(eigs),
  Cumulative = cumsum(eigs)/sum(eigs))

```

```{r overview figure - Da & 16S barplot}

first <- ggarrange(shanSRS_,pca.16,  ncol=2, common.legend = TRUE, legend = "bottom", labels =c("A","B"))


ggarrange(first, fam.16s,  nrow=2, labels=c("","C"))
```

```{r RNA data - read tables - mRNA}

mRNA.counts <- read.delim("Emelie_MetaDenovo/megahit.prokka.counts.tsv", stringsAsFactors = F, header =T)  #smallest sample size 388497 -> 10/388497=very small...
  
mRNA.anno<- read.delim("Emelie_MetaDenovo/megahit.prokka.annotations.tsv", stringsAsFactors = F, header =T)
mRNA.emapper<- read.delim("Emelie_MetaDenovo/megahit.prokka.emapper.tsv", stringsAsFactors = F, header =T)
mRNA.taxa <- read.delim("Emelie_MetaDenovo/megahit.prokka.[].taxonomy_classification.tsv", stringsAsFactors = F, header =T)
mRNA.meta <-  read.table("Emelie_MetaDenovo/meta_rna.new.txt", stringsAsFactors = F, fileEncoding="latin1",sep="\t", header=T)#%>%dplyr::select(-X, -X.1, -X.2, -X.3, -X.4)

#Phytoref blast
phytoref <- read.delim("Emelie_MetaDenovo/prokka_blast_nt.tsv", stringsAsFactors = F, header=F)%>%
  dplyr::rename(chr = V1)%>%
  group_by(chr)%>%
  filter(row_number()==1)%>%
  ungroup()
  

#combine tables Prokaryotes, Archaea with Ribosome related genes fltered out

mRNA.all <- mRNA.counts %>% left_join(mRNA.taxa, by="orf")%>% left_join(mRNA.emapper, by="orf")%>% left_join(mRNA.meta, by="sample")%>%
  #filter(!sample %in% "109")%>%
  anti_join(phytoref, by="chr")%>%
  filter(Domain %in% c("Bacteria", "Archaea"))%>%
  dplyr::filter(!across(Description, ~ grepl('Ribosomal', .)))%>%
  dplyr::filter(!across(Description, ~ grepl('ribosomal', .)))%>%
  dplyr::filter(!across(Description, ~ grepl('ribosomes', .)))%>%
  dplyr::filter(!across(Description, ~ grepl('ribosome', .)))%>%
  dplyr::filter(!across(Preferred_name, ~ grepl('rpl', .)))%>%
   dplyr::filter(!across(Preferred_name, ~ grepl("RPL", .)))%>%
   dplyr::filter(!across(Preferred_name, ~ grepl("rps", .)))%>%
   dplyr::filter(!across(Preferred_name, ~ grepl('RPS', .)))


## KEGG List with different Levels based on KEGG
KEGGlist <- read.delim("Emelie_MetaDenovo/ko00001_ed.txt")
```

```{r Alpha diversity - based on mRNA transcripts}
#SRS analysis
example_input_data.rna <- mRNA.all %>%
  #mRNA.counts %>%
  filter(!count < 5)%>%
  dplyr::select(orf, sample, count)%>%
  spread(sample,count, fill= 0) %>% 
  remove_rownames() %>% 
  column_to_rownames(var = "orf")

#(e.g. species counts of the library with the lowest sequencing depth):
Cmin.rna <- min(colSums(example_input_data.rna))
Cmin.rna

SRS_output.rna <- SRS(data = example_input_data.rna, Cmin = Cmin.rna)
SRS_output.rna

#Shannon Diversity Index
mRNA.meta$sample<-as.character(mRNA.meta$sample)

SRSshannon.rna<- SRS_output.rna %>%
  t()%>%
  data.frame()%>% 
  rownames_to_column(var="sample")%>% 
  plyr::ddply(~sample, function(x) {vegan::diversity(x[-1], index="shannon")}) %>%
  dplyr::rename(shannon= V1) %>% #Shannon 
  left_join(mRNA.meta, by="sample")

##Plot

SRS.RNA <- SRSshannon.rna %>%
   group_by(bay)%>%
  mutate(mean=mean(shannon))%>%
  mutate(sd=sd(shannon))%>%
  ungroup()%>%
 ggplot(aes(x = bay, y = mean, color=bay)) +
  geom_point(size=3) +
   geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.3,
                 position=position_dodge(.05)) +
scale_color_manual(values=cols.bay)+
  guides(size=guide_legend(order=2),
         shape= guide_legend(override.aes=list(size=2),
        colour=guide_legend(override.aes=list(size=4))))+
 labs(x = "bay", y = "Shannon Index") +
  theme_classic() +
  theme(text = element_text(size = 14)) +
  theme(panel.grid = element_blank(), axis.title.x = element_text(size = 12),legend.position = "none")

#Test to compare transcript of bays with each other
kruskal.test(shannon ~ bay, data = SRSshannon.rna )
```

```{r Top Taxa overview - mRNA}
#Top10 Family 

TopN <- 6
Barplot.RNA <- mRNA.all%>%
   filter(!Family %in% c("NA uncl."))%>%
  drop_na(Family)%>%
  group_by(bay, name2, Family)%>%
  summarise(tpm_mean=mean(tpm))%>%
  arrange(desc(tpm_mean))%>%
  mutate(aggTaxo=as.character(case_when(
    row_number()<=TopN~Family, row_number()>TopN~'Others')))%>%
  ggplot(aes(x = name2, y = tpm_mean, fill = aggTaxo)) +
geom_bar( position="fill", stat="identity") +
coord_flip()+
     scale_fill_manual(values=c(  "#84afde", "#7ab5bc","#a2abc9","#e3fffa","#a6cee3","#1f78b4", "#73cfc6", "#ffffcd", "lightgoldenrod",  "#b2df8a", "yellowgreen", "#33a02c", "olivedrab", "#9eb383", "lemonchiffon3", "lightpink",  "#ddcaff", "#d49abe", "#db9d88", "#bea4b6", "#fb9a99", "grey"
  ))+
  theme(
    legend.position = 'bottom'
  )+
  theme(
    axis.text.x = element_text(angle = 60, hjust = 1)
  )+
   labs(x = "bay", y = "tpm (%)") +
    theme_classic() +
  theme(text = element_text(size = 14)) +
  theme(panel.grid = element_blank(), axis.title.x = element_text(size = 12))+
  labs(fill='Family') 

length(unique(mRNA.all$Family))
```

```{r Differential expression analysis - mRNA}
#Table
counts.mRNA.format <- mRNA.all%>%
  dplyr::select(orf, sample, count) %>% 
  spread(sample,count, fill= 0)

counts.mRNA.format <- data.frame(counts.mRNA.format)
(sum(colSums(counts.mRNA.format == 0))) / (nrow(counts.mRNA.format) * ncol(counts.mRNA.format))#40 % Zeros


dds <- DESeqDataSetFromMatrix(countData=counts.mRNA.format, colData=mRNA.meta, design=~bay, tidy=T)


#Prefiltering low abundant counts
smallestGroupSize <- 6
keep <- rowSums(counts(dds) >= 10) >= smallestGroupSize
dds <- dds[keep,]

#Normalized counts for downstream analysis
#dds <- estimateSizeFactors(dds)
#sizeFactors(dds)
#normalized_counts <- counts(dds, normalized=TRUE)

dds <- DESeq(dds, test="LRT", reduced=~1, sfType="poscounts",
                   minReplicatesForReplace=Inf, fitType = "local")

res <- results(dds)
head(results(dds, tidy=T))
summary(res) #37% upregulated, 29% downregulated, 1.7% outlier

res <- res[order(res$padj),]
head(res)
plotMA(dds)
plotDispEsts(dds)  


resultsNames(dds)
bay.res<- results(dds, contrast=c("bay","heated","control")) 


    bay.res<- data.frame(bay.res) %>% rownames_to_column(var ="orf")
    
    baseMeanPerLvl <- sapply( levels(dds$bay), function(lvl) rowMeans(counts(dds,normalized=TRUE)[,dds$bay == lvl] ) )
    baseMeanPerLvl <- as.data.frame(baseMeanPerLvl)%>%
      rownames_to_column(var="orf")
 
    #Final Table
    bay.res  <-bay.res %>% left_join(baseMeanPerLvl, by="orf") 
    bay.res <- bay.res [,c(1,2,8,9,3,4,5,6,7)]
    
```

```{r PCA based on DE-object with log transformated values}
# Input is a matrix of log transformed values
rld <- rlog(dds, blind=T)
rld_mat <- assay(rld)
pca <- prcomp(t(rld_mat))

# Create data frame with metadata and PC3 and PC4 values for input to ggplot
df <- cbind(mRNA.meta, pca$x)
PCA.RNA <- ggplot(df,aes(x=PC1, y=PC2, color = bay, size=10)) + geom_point(size=5)+
   scale_color_manual(values=cols.bay)+
        theme_classic()+
     theme(axis.text.x = element_text(angle = 45, hjust=1),
           panel.background = element_rect(fill="white", color="black"))+
  theme(
    legend.position = 'bottom'
  )+
   theme_classic()+
  theme(
    legend.position = 'bottom'
  )

#Proportion explained 
eigs <- pca$sdev^2
eigs[1] / sum(eigs)

rbind(
  SD = sqrt(eigs),
  Proportion = eigs/sum(eigs),
  Cumulative = cumsum(eigs)/sum(eigs))

```

```{r combine RNA figures}
first <- ggarrange(SRS.RNA,PCA.RNA,  ncol=2, common.legend = TRUE, legend = "bottom", labels =c("A","B"))


ggarrange(first, Barplot.RNA,  nrow=2, labels=c("","C"))
```

```{r SUPP Figure DE expression on Fam level}
bay.res.sig <-bay.res  %>%
        filter(padj <0.05) %>%
  filter(!between(log2FoldChange, -1, 1))%>%
  left_join(mRNA.all, by=c("orf"))

bay.res.sig%>%
  na.omit(Family)%>%
  filter(!Family %in% "-")%>%
ggplot( aes(Phylum,Family, fill=log2FoldChange))+
    geom_tile()+
   # coord_flip()+
    scale_fill_gradient2(low = "blue",
                  mid="white",
                      high = "orange",
                      guide = "colorbar",
                  limits=c(-10,10))+
      theme_classic()+
     theme(axis.text.x = element_text(angle = 45, hjust=1),
           panel.background = element_rect(fill="white", color="black"))+
  labs(y="Family")


```

```{r DE Figures - Stress - mRNA}
bay.res.sig <-bay.res  %>%
        filter(padj <0.05) %>%
  filter(!between(log2FoldChange, -1, 1))%>%
  left_join(mRNA.all, by=c("orf"))

bay.res.sig$KEGG_ko <- gsub("ko:","",bay.res.sig$KEGG_ko)

mRNA.sig.KO.bay <- rename(bay.res.sig, c("KEGG_ko"="KO"))

mRNA.sig.KO.bay <- mRNA.sig.KO.bay %>% inner_join(KEGGlist, by="KO")


mRNA.sig.KO.sel<-mRNA.sig.KO.bay %>% filter(Secondhighest %in% c("09102 Energy metabolism","09182 Protein families: genetic information processing"))%>%
 filter(Thirdhighest %in% "03110 Chaperones and folding catalysts [BR:ko03110]")%>%
  dplyr::select(orf, gene_info)%>%
   group_by(orf)%>%filter(row_number() == 1)%>%ungroup()

mRNA.sig.chap.bay <-bay.res.sig %>% inner_join(mRNA.sig.KO.sel, by=c("orf"))

#Hatmap #positive heated #negative control
mRNA.sig.chap.bay%>%
  na.omit(Preferred_name)%>%
  filter(!Preferred_name %in% "-")%>%
ggplot( aes(reorder(Family, -log2FoldChange),Preferred_name, fill=log2FoldChange))+
    geom_tile()+
    coord_flip()+
    scale_fill_gradient2(low = "blue",
                  mid="white",
                      high = "orange",
                      guide = "colorbar",
                  limits=c(-8,8))+
      theme_classic()+
     theme(axis.text.x = element_text(angle = 45, hjust=1),
           panel.background = element_rect(fill="white", color="black"))+
  labs(y="gene transcript")


```

```{r DE - energy metabolism table - mRNA}
###energy metabolism

mRNA.sig.KO.energy<-mRNA.sig.KO.bay %>% 
  filter(Secondhighest %in% c("09102 Energy metabolism"))%>%
  dplyr::select(orf, tpm, Phylum, Class, Order, Family, Genus, Species, Preferred_name, Description, PFAMs, bay, gene_info, Thirdhighest, log2FoldChange, KO)

##overview about energy metabolism
plot1 <- mRNA.sig.KO.energy%>%
  group_by(gene_info, Thirdhighest)%>%
  summarise(log2FoldChange=mean(log2FoldChange))%>%
  ungroup()%>%
 # na.omit(Preferred_name)%>%
  #filter(!Preferred_name %in% "-")%>%
ggplot( aes(reorder(gene_info, -log2FoldChange),Thirdhighest, fill=log2FoldChange))+
    geom_tile()+
    #coord_flip()+
    scale_fill_gradient2(low = "blue",
                  mid="white",
                      high = "orange",
                      guide = "colorbar",
                  limits=c(-10,10))+
      theme_classic()+
     theme(
       #axis.title.x=element_blank(),
       # axis.text.x=element_blank(),
      #  axis.ticks.x=element_blank(),
           axis.text.y = element_text(angle = 0, hjust=1),
           axis.text.x = element_text(angle = 45, hjust=1),
           panel.background = element_rect(fill="white", color="black"))+
  labs(x="gene transcript", y="energy metabolism")+
     scale_y_discrete(label = function(x) stringr::str_trunc(x, 35))+
  scale_x_discrete(label = function(x) stringr::str_trunc(x, 18))




sum.gene <- mRNA.sig.KO.energy %>%
  group_by(bay, gene_info, Thirdhighest)%>%
  summarise(log2FoldChange =mean(log2FoldChange))%>%
  ungroup()

plot2 <- mRNA.sig.KO.energy%>%
    group_by(gene_info, Family)%>%
  summarise(log2FoldChange=mean(log2FoldChange))%>%
  ungroup()%>%
 # na.omit(Preferred_name)%>%
  #filter(!Preferred_name %in% "-")%>%
ggplot( aes(gene_info,Family, fill=log2FoldChange))+
    geom_tile()+
   # coord_flip()+
    scale_fill_gradient2(low = "blue",
                  mid="white",
                      high = "orange",
                      guide = "colorbar",
                  limits=c(-10,10))+
      theme_classic()+
     theme(axis.text.x = element_text(angle = 45, hjust=1),
           axis.text.y = element_text(angle=0, hjust=1),
           panel.background = element_rect(fill="white", color="black"))+
  labs(x="gene transcript", y="Family")+
   scale_y_discrete(label = function(x) stringr::str_trunc(x, 35))+
  scale_x_discrete(label = function(x) stringr::str_trunc(x, 18))



ggarrange(plot1, plot2, nrow=2, common.legend = TRUE, legend="right",labels =c("A","B"),heights = c(4, 4), align = "v")
  
```

```{r Families Overview table 16S and mRNA data}
Genus.16S<- Tab_all %>% 
  group_by(Genus, bay, sample)%>% 
  summarise(relab= sum(relab))%>%
  summarise(sum_relab= mean(relab))%>%
  filter(!sum_relab<0.0005)%>%
  ungroup()


mRNA.all%>%
  group_by(Family, bay)%>%
  summarise(tpm=mean(tpm))%>%
  ungroup()%>%
  filter(tpm >4)%>%
  arrange()
```

```{r session-info}
# Display current R session information
sessionInfo()
```
